docs/rename_function.py

```diff
@@ -0,0 +1,74 @@
+"""
+This script requires developers to add the following information:
+1. add file and directory name to srcdir_srcfiles
+2. add file and function name to srcfiles_srcfuncs
+3. add expected display name for the function to display_names
+"""
+
+
+
+import os
+import itertools
+from shutil import copyfile
+
+# Add the function names with the src file
+# srcfiles_srcfuncs = { srcfile : [func1, func2..]}
+# key : srcfile
+# values per key : function names that should be changed
+# If the file is already present, just add the func name in the respective values
+# Create new entry if the srcfile is not present
+srcfiles_srcfuncs = {
+    "hpat_pandas_series_functions.py" : ["hpat_pandas_series_append", "hpat_pandas_series_ne", "hpat_pandas_series_iloc"]
+}
+
+
+# Add the filenames and the parent directory
+# srcdir_srcfiles = { parentdir : [filename1, filename2..]}
+# key: parent dirname
+# values per key : filenames to be copied within the parent dir
+# If the dir is already present in this list, just add the filename in the respective values
+# Create a new entry if the dir is not present in this dictionary
+srcdir_srcfiles = {
+    "hpat/datatypes" : ["hpat_pandas_series_functions.py"],
+    "hpat/hiframes"  : ["aggregate.py", "boxing.py"]
+}
+
+
+# Add the function name that will replace the original name and should be displayed in documentation
+# Always add new name at the ends. Do not change the order
+display_names = ['append', 'ne', 'iloc']
+cur_dir = os.getcwd()
+
+# This is the dir where all the source files will be copied
+src_copy_dir = os.path.join(cur_dir, "..", "API_Doc")
+if not os.path.exists(src_copy_dir):
+    os.mkdir(src_copy_dir)
+
+# Copy all required srcfiles
+for dir in srcdir_srcfiles:
+    file_list = srcdir_srcfiles[dir]
+    for f in file_list:
+        src_file = os.path.join(cur_dir, "..", dir, f)
+        dst_file = os.path.join(cur_dir, "..", "API_Doc", f)
+        copyfile(src_file, dst_file)
+        
+os.chdir(src_copy_dir)
+
+# Change the function names in copied files
+i = 0
+for filename in srcfiles_srcfuncs:
+    func_list = srcfiles_srcfuncs[filename]
+    with open(filename, '+r') as fn:

```

Comment:
please don't forget to close file

Reply:
That is the advantage of using "with open". It will close the file automatically.

Reply:
I guess the 'usual' way to do this is to end a `with open('r')` clause after reading the file and open a new `with open('w')` clause for writing.

Reply:
@fschlimb Yes that's another way. Apart from the codestyle I don't see any other difference in both approach. So @shssf if you feel more comfortable with this second approach I can change this code. What do you think?